database_cleanup/tests/test_database_cleanup.py

```diff
@@ -78,3 +89,9 @@ def test_database_cleanup(self):
         with self.assertRaises(ProgrammingError):
             with self.registry.cursor() as cr:
                 self.env.cr.execute('select * from database_cleanup_test')
+        with self.registry.cursor() as cr2:
+            # Release the delete of ir_module_module pending
+            self.env.cr.rollback()
+            cr2.execute(
+                "DELETE FROM ir_module_module WHERE id=%s", (module.id,))
```

Comment:
If you run 2 times this test, you will a contraint erorr of ir_module_module name unique.
It's because the process: `self.env['ir.module.module'].create(...)` and `RegistryManager.registries[self.env.cr.dbname] = original_registry` have a custom cursor with commit (different to current `self.env.cr`)

Then the test purge using `self.env.cr` and work fine because that cursor have a delete.
But all tests use `rollback` at final.
Then you will have a dirty database with a module called: `database_cleanup_test`

I mean, without this patch
 - `cursor_x`

```bash
BEGIN;
INSERT INTO ir_module_module
COMMIT;
END;
```

- `self.env.cr`
```bash
BEGIN;
DELETE FROM ir_module_module
ROLLBACK;
END;
```